src/libraries/Microsoft.PowerFx.Interpreter/PatchFunction.cs

```diff
@@ -0,0 +1,187 @@
+            {
+                fieldList.Add(new NamedValue(field.Key, field.Value));
+            }
+
+            var record = RecordValue.NewRecordFromFields(fieldList);
+
+            return Task.FromResult<FormulaValue>(record);
+        }
+
+        public override RequiredDataSourcePermissions FunctionPermission => RequiredDataSourcePermissions.Create | RequiredDataSourcePermissions.Update;
+    }
+
+    internal class PatchFunction : PatchAndValidateRecordFunctionBase, IAsyncTexlFunction
+    {
+        public override bool IsSelfContained => false;
+
+        public PatchFunction()
+            : base("Patch", TexlStrings.AboutPatch, FunctionCategories.Table | FunctionCategories.Behavior, DType.EmptyRecord, 0, 3, int.MaxValue, DType.EmptyTable, DType.EmptyRecord, DType.EmptyRecord)
+        {
+        }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
+        {
+            yield return new[] { TexlStrings.PatchDataSourceArg, TexlStrings.PatchBaseRecordArg };
+            yield return new[] { TexlStrings.PatchDataSourceArg, TexlStrings.PatchBaseRecordArg, TexlStrings.PatchChangeRecordsArg };
+            yield return new[] { TexlStrings.PatchDataSourceArg, TexlStrings.PatchBaseRecordArg, TexlStrings.PatchChangeRecordsArg, TexlStrings.PatchChangeRecordsArg };
+        }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures(int arity)
+        {
+            if (arity > 3)
+            {
+                return GetGenericSignatures(arity, TexlStrings.PatchDataSourceArg, TexlStrings.PatchBaseRecordArg, TexlStrings.PatchChangeRecordsArg);
+            }
+
+            return base.GetSignatures(arity);
+        }
+
+        public async Task<FormulaValue> InvokeAsync(FormulaValue[] args, CancellationToken cancel)
+        {
+            var argFields = new Dictionary<string, FormulaValue>();
+
+            var arg0 = args[0] as TableValue;
+            var arg1 = args[1] as RecordValue;
+
+            // Starts from the 3rd argument (new values)
+            for (var i = 2; i < args.Length; i++)
+            {
+                var arg = args[i];
+
+                if (arg is BlankValue)
+                {
+                    continue;
+                }
+                else if (arg is ErrorValue)
+                {
+                    // @@@JYL How to handle error value arguments?
+                    return arg;
+                }
+                else if (arg is RecordValue record)
+                {
+                    // @@@JYL What is a ellegant way to avoid the nested loop?
+                    foreach (var field in record.Fields)
+                    {
+                        argFields[field.Name] = field.Value;
+                    }
+                }
+            }
+
+            foreach (KeyValuePair<string, FormulaValue> kvp in argFields)
+            {
+                arg1.UpdateField(new NamedValue(kvp));
+            }
```

Comment:
UpdateField() will be a network call - so we don't want to call that in a loop. 

Instead, could we just change the signature to UpdateField to allow an IEnum<KeyValuePair<string, FormulaValue>> ? 
Then it's 1 network call. 

It can still have other simpler overloads that chain to that.